datahub-web-react/src/app/entityV2/shared/EntitySearchSelect/ActorsSearchSelect.tsx

```diff
@@ -0,0 +1,239 @@
+
+        // Only update if the URNs have actually changed
+        if (currentSelectedUrns !== newSelectedUrns) {
+            const entities = selectedActorUrns
+                .map((urn) => {
+                    // Try to find in all available sources
+                    return (
+                        (placeholderActors || []).find((e) => e.urn === urn) ||
+                        searchResults.find((e) => e.urn === urn) ||
+                        selectedActorEntities.find((e) => e.urn === urn)
+                    );
+                })
+                .filter(Boolean) as Entity[];
+
+            setSelectedActorEntities(entities);
+        }
+    }, [selectedActorUrns, placeholderActors, searchResults, selectedActorEntities]);
+
+    // Use utility to deduplicate entities from all sources
+    const allOptionsEntities = deduplicateEntities({
+        placeholderEntities: placeholderActors,
+        searchResults,
+        selectedEntities: selectedActorEntities,
+    });
+
+    // Convert entities to SelectOption format using utility
+    const selectOptions = entitiesToSelectOptions(allOptionsEntities, entityRegistry);
+
+    // Handle search
+    const handleSearch = useCallback(
+        (query: string) => {
+            if (query.trim()) {
+                autoCompleteQuery({
+                    variables: {
+                        input: {
+                            types: [EntityType.CorpUser, EntityType.CorpGroup],
+                            query: query.trim(),
+                            limit: 10,
+                        },
+                    },
+                });
+            }
+        },
+        [autoCompleteQuery],
+    );
+
+    // Render actor entity in dropdown
+    const renderDropdownActorLabel = useCallback(
+        (entity: Entity) => {
+            const displayName = entityRegistry.getDisplayName(entity.type, entity);
+            const subtitle = entity.type === EntityType.CorpUser ? (entity as any)?.properties?.email : undefined;
+            return (
+                <IconAndNameContainer>
+                    <IconWrapper>
+                        <EntityIcon entity={entity} />
+                    </IconWrapper>
+                    <TitleContainer>
+                        <Text type="div">{displayName}</Text>
+                        {subtitle && (
+                            <Text type="div" size="sm" color="gray">
+                                {subtitle}
+                            </Text>
+                        )}
+                    </TitleContainer>
+                </IconAndNameContainer>
+            );
+        },
+        [entityRegistry],
+    );
+
+    // Render selected actor label
+    const renderSelectedActorLabel = useCallback(
+        (selectedOption: OptionType) => {
+            const entity = allOptionsEntities.find((e) => e.urn === selectedOption.value) as
+                | Partial<CorpUser>
+                | Partial<CorpGroup>;
```

Comment:
i know we're trying to avoid any type casting if possible

Reply:
Trying to. Not sure how to best do this. Kinda a mess